internal/services/eventgrid/eventgrid_partner_registration.go

```diff
@@ -0,0 +1,194 @@
+package eventgrid
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/eventgrid/2022-06-15/partnerregistrations"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tags"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = EventGridPartnerRegistrationResource{}
+
+type EventGridPartnerRegistrationResource struct{}
+
+type EventGridPartnerRegistrationResourceModel struct {
+	Name                  string            `tfschema:"name"`
+	ResourceGroup         string            `tfschema:"resource_group_name"`
+	PartnerRegistrationID string            `tfschema:"partner_registration_id"`
+	Tags                  map[string]string `tfschema:"tags"`
+}
+
+func (EventGridPartnerRegistrationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": &schema.Schema{
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
Should we have a more robust validation function here instead of just `StringIsNotEmpty`? Something like the below:

* **Length:** 3 to 64 characters.
* **Allowed characters:** Alphanumeric characters (a-z, A-Z, 0-9), hyphens (-), and underscores (_).
* **First and last character:** Must be alphanumeric.